pkg/provider/manager.go

```diff
@@ -38,16 +40,39 @@ type nutanixManager struct {
 	client         clientset.Interface
 	config         config.Config
 	nutanixClient  interfaces.Client
-	ignoredNodeIPs map[string]struct{}
+	ignoredNodeIPs *netipx.IPSet
 }
 
 func newNutanixManager(config config.Config) (*nutanixManager, error) {
 	klog.V(1).Info("Creating new newNutanixManager")
 
-	// Initialize the ignoredNodeIPs map
-	ignoredNodeIPs := make(map[string]struct{}, len(config.IgnoredNodeIPs))
+	// Initialize the ignoredNodeIPs set
+	ignoredIPsBuilder := netipx.IPSetBuilder{}
 	for _, ip := range config.IgnoredNodeIPs {
-		ignoredNodeIPs[ip] = struct{}{}
+		switch {
+		case strings.Contains(ip, "-"):
+			ipRange, err := netipx.ParseIPRange(ip)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse ignoredNodeIPs IP range %q: %v", ip, err)
+			}
+			ignoredIPsBuilder.AddRange(ipRange)
+		default:
```

Comment:
The partial use of switch and then another conditional in the default clause seems a bit off to me. Let's either switch to all conditionals or add another case clause for prefix.

Reply:
Updated.